packages/backend-common/src/reading/BitbucketUrlReader.ts

```diff
@@ -45,17 +45,25 @@ import {
  * as the one exposed by Bitbucket Cloud itself.
  *
  * @public
+ * @deprecated in favor of BitbucketCloudUrlReader and BitbucketServerUrlReader
  */
 export class BitbucketUrlReader implements UrlReader {
```

Comment:
Another discussion for a different thread that I pretty arbitrarily chose to but here: I'm noticing this is pretty much only additions. Would it be possible to express more of the existing and deprecated Bitbucket implementations in terms of the two new ones? For example having the existing `BitbucketUrlReader` leverage the two new readers instead of having its own separate implementation?

If that just complicates things unnecessarily I think we should be pretty aggressive about the deprecations so that we can get the old code sooner rather than later. I'm thinking things like logging a warning when `bitbucket` is used.

Reply:
logging a warning sounds like a great idea! It might be possible to reuse the new ones at the old one. I can look into it.

Reply:
ah.. reusing will be tricky due to the selection of a ready by host. But I will add the warning.

Reply:
I added warnings at the UrlReader (and therefore, implicitly the integration) as well as the scaffolder action.